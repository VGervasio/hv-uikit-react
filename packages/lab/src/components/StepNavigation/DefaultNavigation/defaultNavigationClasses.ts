import { getClasses } from "@hitachivantara/uikit-react-core";

export type HvDefaultNavigationClasses = {
  root: string;
};

const classKeys: string[] = ["root"];

const defaultNavigationClasses = getClasses<HvDefaultNavigationClasses>(
  classKeys,
  "HvStepNavigation"
);

export default defaultNavigationClasses;
