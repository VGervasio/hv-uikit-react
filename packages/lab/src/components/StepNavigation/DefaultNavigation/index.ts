export * from "./DefaultNavigation";
export * from "./Step";
export { default as defaultNavigationClasses } from "./defaultNavigationClasses";
export * from "./defaultNavigationClasses";
export * from "./utils";
