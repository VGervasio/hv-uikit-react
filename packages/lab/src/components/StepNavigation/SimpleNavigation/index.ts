export * from "./SimpleNavigation";
export * from "./Dot";
export { default as simpleNavigationClasses } from "./simpleNavigationClasses";
export * from "./simpleNavigationClasses";
export * from "./utils";
