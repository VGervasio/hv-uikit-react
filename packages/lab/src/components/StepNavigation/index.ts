export { default as stepNavigationClasses } from "./stepNavigationClasses";
export * from "./stepNavigationClasses";
export * from "./StepNavigation";
