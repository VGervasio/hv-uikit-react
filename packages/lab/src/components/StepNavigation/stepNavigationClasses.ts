import { getClasses } from "@hitachivantara/uikit-react-core";

export type HvStepNavigationClasses = {
  /** Styles applied to the component root class. */
  root?: string;
  /** Styles applied to the title element. */
  titles?: string;
  /** Styles applied to the li element. */
  li?: string;
  /**  Styles applied to the ol element. */
  ol?: string;
  /** Styles applied to the separator element. */
  separator?: string;
};

const classKeys: string[] = ["root", "li", "ol", "separator", "titles"];

const stepNavigationClasses = getClasses<HvStepNavigationClasses>(
  classKeys,
  "HvStepNavigation"
);

export default stepNavigationClasses;
