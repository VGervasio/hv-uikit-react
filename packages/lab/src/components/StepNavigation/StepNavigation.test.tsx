import { act, render, waitForElementToBeRemoved } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { describe, expect, it } from "vitest";
import { HvStepNavigation } from "./StepNavigation";
import { HvStepProps } from "./DefaultNavigation";

type StepType = Pick<
  HvStepProps,
  "state" | "title" | "onClick" | "className" | "disabled"
> & {
  separatorClassName?: string;
  titleClassName?: string;
};

const steps: StepType[] = [
  {
    title: "Completed",
    state: "Completed",
    onClick: () => {
      alert("You clicked on 'Completed' step");
    },
  },
  {
    title: "Failed",
    state: "Failed",
    onClick: () => {
      alert("You clicked on 'Failed' step");
    },
  },
  {
    title: "Pending",
    state: "Pending",
    onClick: () => {
      alert("You clicked on 'Pending' step");
    },
  },
  {
    title: "Current",
    state: "Current",
    onClick: () => {
      alert("You clicked on 'Current' step");
    },
  },
  {
    title: "Enabled",
    state: "Enabled",
    onClick: () => {
      alert("You clicked on 'Enabled' step");
    },
  },
  {
    title: "Disabled",
    state: "Disabled",
    onClick: () => {
      alert("You clicked on 'Disabled' step");
    },
  },
];

describe("StepNavigation", () => {
  it("should be defined", () => {
    const { container } = render(<HvStepNavigation steps={steps} />);
    expect(container).toBeDefined();
  });

  it("should render correctly", () => {
    const { container } = render(<HvStepNavigation steps={steps} />);
    expect(container).toMatchSnapshot();
  });

  it("should contain all the steps", () => {
    const { getByLabelText } = render(<HvStepNavigation steps={steps} />);

    expect(getByLabelText("step-Completed")).toBeInTheDocument();
    expect(getByLabelText("step-Failed")).toBeInTheDocument();
    expect(getByLabelText("step-Pending")).toBeInTheDocument();
    expect(getByLabelText("step-Current")).toBeInTheDocument();
    expect(getByLabelText("step-Enabled")).toBeInTheDocument();
    expect(getByLabelText("step-Disabled")).toBeInTheDocument();
  });

  it("should show a tooltip on hover", () => {
    const { getByLabelText, findByRole, queryByRole } = render(
      <HvStepNavigation type="Simple" showTitles={false} steps={steps} />
    );
    act(async () => {
      userEvent.hover(getByLabelText("step-container-Pending"));
      const tooltip = await findByRole("tooltip");
      expect(tooltip).toBeInTheDocument();
      userEvent.unhover(getByLabelText("step-container-Pending"));
      await waitForElementToBeRemoved(() => queryByRole("tooltip"));
      expect(tooltip).not.toBeInTheDocument();
    });
  });
});
