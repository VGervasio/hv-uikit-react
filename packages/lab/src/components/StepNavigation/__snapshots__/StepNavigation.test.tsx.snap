// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StepNavigation > should render correctly 1`] = `
<div>
  <div>
    <nav
      style="width: 1390px; margin: 0px;"
    >
      <ol
        class="css-fg3xwr HvStepNavigation-ol"
      >
        <li
          class="HvStepNavigation-li css-6715mb-styledLi e1wrrltm3"
        >
          <div
            class="HvStep-root HvStep-notCurrent css-15wpxis-StyledRoot e1bnfr852"
          >
            <button
              aria-label="step-Completed"
              class="HvStep-ghost e1bnfr851 HvButton-root css-u7l3xp-StyledButton"
            >
              <div
                class="css-4uy5a6"
              >
                <span
                  class="css-hg21i6"
                >
                  <div
                    class="HvAvatar-container css-mtgipz"
                  >
                    <div
                      class="HvAvatar-status HvAvatar-circular HvAvatar-sm css-zck2nt"
                    >
                      <div
                        class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault HvStep-avatar HvStep-sm e1bnfr850 HvAvatar-root HvAvatar-avatar HvAvatar-sm css-1ns3t76-StyledAvatar css-2s90m6-MuiAvatar-root"
                      >
                        <div
                          class="css-f95wjz"
                          name="Level0Good"
                        >
                          <svg
                            focusable="false"
                            height="24"
                            viewBox="0 0 24 24"
                            width="24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <circle
                              class="color0"
                              cx="12"
                              cy="12"
                              fill="var(--uikit-colors-positive)"
                              r="12"
                            />
                            <path
                              class="color1"
                              d="m9.95 17.1-3.7-3.8 1.4-1.4 2.3 2.2 6.2-7.2 1.6 1.4z"
                              fill="var(--uikit-colors-atmo2)"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                  </div>
                </span>
              </div>
            </button>
          </div>
        </li>
        <li
          aria-hidden="true"
          class="HvStepNavigation-separator css-xc81e9-StyledLi e1wrrltm1"
        >
          <div
            aria-label="separator-Completed"
            class=""
          />
        </li>
        <li
          class="HvStepNavigation-li css-6715mb-styledLi e1wrrltm3"
        >
          <div
            class="HvStep-root HvStep-notCurrent css-15wpxis-StyledRoot e1bnfr852"
          >
            <button
              aria-label="step-Failed"
              class="HvStep-ghost e1bnfr851 HvButton-root css-u7l3xp-StyledButton"
            >
              <div
                class="css-4uy5a6"
              >
                <span
                  class="css-hg21i6"
                >
                  <div
                    class="HvAvatar-container css-mtgipz"
                  >
                    <div
                      class="HvAvatar-status HvAvatar-circular HvAvatar-sm css-zck2nt"
                    >
                      <div
                        class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault HvStep-avatar HvStep-sm e1bnfr850 HvAvatar-root HvAvatar-avatar HvAvatar-sm css-1ns3t76-StyledAvatar css-2s90m6-MuiAvatar-root"
                      >
                        <div
                          class="css-f95wjz"
                          name="Level3Bad"
                        >
                          <svg
                            focusable="false"
                            height="24"
                            viewBox="0 0 24 24"
                            width="24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              class="color0"
                              d="M12 0A12 12 0 1 1 0 12 12 12 0 0 1 12 0z"
                              fill="var(--uikit-colors-negative)"
                            />
                            <path
                              class="color1"
                              d="M10.88 15.81h2.25v2.25h-2.25zm0-9.87h2.25v8h-2.25z"
                              fill="var(--uikit-colors-atmo2)"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                  </div>
                </span>
              </div>
            </button>
          </div>
        </li>
        <li
          aria-hidden="true"
          class="HvStepNavigation-separator css-19ukis7-StyledLi e1wrrltm1"
        >
          <div
            aria-label="separator-Failed"
            class=""
          />
        </li>
        <li
          class="HvStepNavigation-li css-6715mb-styledLi e1wrrltm3"
        >
          <div
            class="HvStep-root HvStep-notCurrent css-15wpxis-StyledRoot e1bnfr852"
          >
            <button
              aria-label="step-Pending"
              class="HvStep-ghost e1bnfr851 HvButton-root css-u7l3xp-StyledButton"
            >
              <div
                class="css-4uy5a6"
              >
                <span
                  class="css-hg21i6"
                >
                  <div
                    class="HvAvatar-container css-mtgipz"
                  >
                    <div
                      class="HvAvatar-status HvAvatar-circular HvAvatar-sm css-zck2nt"
                    >
                      <div
                        class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault HvStep-avatar HvStep-sm e1bnfr850 HvAvatar-root HvAvatar-avatar HvAvatar-sm css-1ns3t76-StyledAvatar css-2s90m6-MuiAvatar-root"
                      >
                        <div
                          class="css-f95wjz"
                          name="HourGlass"
                        >
                          <svg
                            focusable="false"
                            height="16"
                            viewBox="0 0 16 16"
                            width="16"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              class="color0"
                              d="M13.028 1H15V0H1v1h2.01v4.268L7.116 8l-4.1 2.73V15H1v1h14v-1h-1.975v-4.269L8.922 8l4.106-2.733zm-1 10.269V15H4.013v-3.731L8.019 8.6zm0-6.537L8.019 7.4 4.01 4.732V1h8.018z"
                              fill="var(--uikit-colors-atmo2)"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                  </div>
                </span>
              </div>
            </button>
          </div>
        </li>
        <li
          aria-hidden="true"
          class="HvStepNavigation-separator css-gstx76-StyledLi e1wrrltm1"
        >
          <div
            aria-label="separator-Pending"
            class=""
          />
        </li>
        <li
          class="HvStepNavigation-li css-1g1e0xp-styledLi e1wrrltm3"
        >
          <div
            class="HvStep-root css-1668iss-StyledRoot e1bnfr852"
          >
            <button
              aria-label="step-Current"
              class="HvStep-ghost HvStep-ghostDisabled e1bnfr851 HvButton-root css-jfzapv-StyledButton"
              disabled=""
            >
              <div
                class="css-4uy5a6"
              >
                <span
                  class="css-hg21i6"
                >
                  <div
                    class="HvAvatar-container css-mtgipz"
                  >
                    <div
                      class="HvAvatar-status HvAvatar-circular HvAvatar-sm css-zck2nt"
                      style="box-shadow: inset 0px 0px 0px 2px var(--uikit-colors-secondary_60);"
                    >
                      <div
                        class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault HvStep-avatar HvStep-sm e1bnfr850 HvAvatar-root HvAvatar-avatar HvAvatar-sm css-1ns3t76-StyledAvatar css-2s90m6-MuiAvatar-root"
                      >
                        4
                      </div>
                    </div>
                  </div>
                </span>
              </div>
            </button>
          </div>
        </li>
        <li
          aria-hidden="true"
          class="HvStepNavigation-separator css-19ch03f-StyledLi e1wrrltm1"
        >
          <div
            aria-label="separator-Current"
            class=""
          />
        </li>
        <li
          class="HvStepNavigation-li css-6715mb-styledLi e1wrrltm3"
        >
          <div
            class="HvStep-root HvStep-notCurrent css-15wpxis-StyledRoot e1bnfr852"
          >
            <button
              aria-label="step-Enabled"
              class="HvStep-ghost e1bnfr851 HvButton-root css-u7l3xp-StyledButton"
            >
              <div
                class="css-4uy5a6"
              >
                <span
                  class="css-hg21i6"
                >
                  <div
                    class="HvAvatar-container css-mtgipz"
                  >
                    <div
                      class="HvAvatar-status HvAvatar-circular HvAvatar-sm css-zck2nt"
                    >
                      <div
                        class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault HvStep-avatar HvStep-sm e1bnfr850 HvAvatar-root HvAvatar-avatar HvAvatar-sm css-1ns3t76-StyledAvatar css-2s90m6-MuiAvatar-root"
                      >
                        5
                      </div>
                    </div>
                  </div>
                </span>
              </div>
            </button>
          </div>
        </li>
        <li
          aria-hidden="true"
          class="HvStepNavigation-separator css-ss3ju7-StyledLi e1wrrltm1"
        >
          <div
            aria-label="separator-Enabled"
            class=""
          />
        </li>
        <li
          class="HvStepNavigation-li css-6715mb-styledLi e1wrrltm3"
        >
          <div
            class="HvStep-root HvStep-notCurrent css-15wpxis-StyledRoot e1bnfr852"
          >
            <button
              aria-label="step-Disabled"
              class="HvStep-ghost e1bnfr851 HvButton-root css-jfzapv-StyledButton"
              disabled=""
            >
              <div
                class="css-4uy5a6"
              >
                <span
                  class="css-hg21i6"
                >
                  <div
                    class="HvAvatar-container css-mtgipz"
                  >
                    <div
                      class="HvAvatar-status HvAvatar-circular HvAvatar-sm css-zck2nt"
                    >
                      <div
                        class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault HvStep-avatar HvStep-sm e1bnfr850 HvAvatar-root HvAvatar-avatar HvAvatar-sm css-1ns3t76-StyledAvatar css-2s90m6-MuiAvatar-root"
                      >
                        6
                      </div>
                    </div>
                  </div>
                </span>
              </div>
            </button>
          </div>
        </li>
      </ol>
    </nav>
    <div
      class="css-1t1ny2y HvStepNavigation-titles"
    >
      <label
        class="e1wrrltm0 HvTypography-root HvTypography-label css-1jvdi05-StyledTitle"
      >
        Completed
      </label>
      <label
        class="e1wrrltm0 HvTypography-root HvTypography-label css-1jvdi05-StyledTitle"
      >
        Failed
      </label>
      <label
        class="e1wrrltm0 HvTypography-root HvTypography-label css-1jvdi05-StyledTitle"
      >
        Pending
      </label>
      <label
        class="e1wrrltm0 HvTypography-root HvTypography-label css-1jvdi05-StyledTitle"
      >
        Current
      </label>
      <label
        class="e1wrrltm0 HvTypography-root HvTypography-label css-1jvdi05-StyledTitle"
      >
        Enabled
      </label>
      <label
        class="e1wrrltm0 HvTypography-root HvTypography-label css-xhat2k-StyledTitle"
      >
        Disabled
      </label>
    </div>
  </div>
</div>
`;
